lib/spack/spack/all_commands.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""Utility functions relevant to all commands, including Spack internal
+commands and extension commands provided by third party packages.
+"""
+
+import spack.cmd
+import spack.extensions
+
+# Cache found spack commands.
+_all_commands = None
+
+
+def all_commands():
```

Comment:
That would reintroduce the circularity because `all_commands()` calls `spack.extensions.get_command_paths()` from `spack.extensions`, which in turn requires `spack.cmd`.